### ISSUE TITLE
Add translator field to article content type to platform?

### ISSUE THREAD

**Issue by pyrello** [uiowa/uiowa#5414]
This would be a entity reference field set up like the author field.

**Comment by joewhitsitt** [comment]
Unless this becomes a bigger slice to incorporate more of https://schema.org/Article, IMO, this seems like site specific. Perhaps using the body to capture this and then making sure folks are trained to fill the SEO settings schema Article fields correctly if you don't think splitting off an additional field is worth it.

**Comment by pyrello** [comment]
The reason this is being considered this way is because adding an additional field to the article content type will mean that the layout builder layout will be different and will require another full config split of `core.entity_view_display.node.article.default`. This is a maintenance burden for the team over time as we make changes to the article content type. This doesn't happen as much as with the page content type, but is worth considering.

@joewhitsitt I'm not sure why this field has to be considered as part of a larger all-or-nothing addition of fields. If we are getting nominal requests for other fields in the article schema, I think we should consider them. But I don't think the addition of any of them needs to be contingent on adding all of them.

**Comment by joewhitsitt** [comment]
With CS2 this is a smaller burden than it used to be and the use-case for just a translator field doesn't fit the 80% in my opinion. If anything, providing a translation and wiring that up to supplement the article sounds more beneficial. 

That said, the notable missing pieces I see that more folks would make use of are author and publisher which are captured with existing fields but not wired up to our schema config.

I'd be curious if @mjoneill had other thoughts about what we are missing in terms of Article schema.

Thinking about this more, figuring out a way to use the author field to capture individual roles associated with the article (author, collaborator, editor, photographer, point of contact, translator, etc.) seems like a better default offering (though complex).

I am willing to back down on this, but these are my initial thoughts.

**Comment by pyrello** [comment]
> With CS2 this is a smaller burden than it used to be and the use-case for just a translator field doesn't fit the 80% in my opinion. If anything, providing a translation and wiring that up to supplement the article sounds more beneficial.

CS2 does not make this particular case less of a burden. Otherwise, we wouldn't be having this conversation. This is one of the outlier cases that requires a full split of the config object because of sections not having UUIDs. We can't patch LB templates because they don't get merged together correctly.

I certainly can appreciate the 80% philosophy as a good starting point for thinking about whether something is worth the investment. However, I think that using the 80% condition as a way of not adding features because they don't have broad level usage doesn't necessarily seem like the right heuristic to me. The amount of work that will be done to add this field will be the same regardless of whether it is added to the platform or just the site (possibly slightly more because of splitting it), so it is not a question of whether we will make the investment. Does the proposed feature significantly make things harder or worse for the 80%? I'm not sure how to answer this question, but it seems like it is mostly something that people would ignore.

> Thinking about this more, figuring out a way to use the author field to capture individual roles associated with the article (author, collaborator, editor, photographer, point of contact, translator, etc.) seems like a better default offering (though complex).

This is an intriguing idea (and perhaps something that could be adapted to replace what was created on UI Press eventually as a bunch of separate field). In the meantime, what I would be interested in doing is creating a path that allows us to eventually update hook our way into that future.

**Comment by mjoneill** [comment]
> I'd be curious if @mjoneill had other thoughts about what we are missing in terms of Article schema.

Oy...  well, let's start with "translator".  To me, that seems like a good fit for an academic institution, as translation is a line of academic inquiry, etc.  While not every site will need to make use of that field, I could see that potential being important for comparative literature, classics, many literature-based programs, and certainly other disciplines.  

Other terms that describe a person in a professional discipline that are in the schema vocab that might be of interest for Article markup include:

- author(already covered)
- accountablePerson 
- contributor
- copyrightHolder
- creator
- editor
- funder 
- maintainer 
- producer
- provider 
- publisher
- sponsor

Note that for many of the above, the term may be described as either a person or an organization. 

I would NOT recommend we go ahead and add all these elements for use in article type schema if they are not already available to us in a module... my assumption is that we're exposing a subset of the available schema elements to an editor in the UI.  (Is this a fair assumption?) 

How can I best see what elements are available to us in the module to use?

**Comment by mjoneill** [comment]
Additional note: You can see the schema elements for Article here: https://schema.org/Article. There are more specific article types that may be appropriate: 

- [AdvertiserContentArticle](https://schema.org/AdvertiserContentArticle)
- [NewsArticle](https://schema.org/NewsArticle)
- [Report](https://schema.org/Report)
- [SatiricalArticle](https://schema.org/SatiricalArticle)
- [ScholarlyArticle](https://schema.org/ScholarlyArticle)
- [SocialMediaPosting](https://schema.org/SocialMediaPosting)
- [TechArticle](https://schema.org/TechArticle) 

And lastly, it is worth mentioning that Article itself is a specific type of [CreativeWork](https://schema.org/CreativeWork), of which there are many more-specific schemas. Some of these we are already trying to create content types for (or have created them already)...my guess is without the corresponding schema. In most cases, that's probably just fine...but it's worth considering if there are some that might be useful in our ecosystem or on some of our specific sites:

- [AmpStory](https://schema.org/AmpStory)
- [ArchiveComponent](https://schema.org/ArchiveComponent)
- [Article](https://schema.org/Article)
- [Atlas](https://schema.org/Atlas)
- [Blog](https://schema.org/Blog)
- [Book](https://schema.org/Book)
- [Chapter](https://schema.org/Chapter)
- [Claim](https://schema.org/Claim)
- [Clip](https://schema.org/Clip)
- [Collection](https://schema.org/Collection)
- [ComicStory](https://schema.org/ComicStory)
- [Comment](https://schema.org/Comment)
- [Conversation](https://schema.org/Conversation)
- [Course](https://schema.org/Course)
- [CreativeWorkSeason](https://schema.org/CreativeWorkSeason)
- [CreativeWorkSeries](https://schema.org/CreativeWorkSeries)
- [DataCatalog](https://schema.org/DataCatalog)
- [Dataset](https://schema.org/Dataset)
- [DefinedTermSet](https://schema.org/DefinedTermSet)
- [Diet](https://schema.org/Diet)
- [DigitalDocument](https://schema.org/DigitalDocument)
- [Drawing](https://schema.org/Drawing)
- [EducationalOccupationalCredential](https://schema.org/EducationalOccupationalCredential)
- [Episode](https://schema.org/Episode)
- [ExercisePlan](https://schema.org/ExercisePlan)
- [Game](https://schema.org/Game)
- [Guide](https://schema.org/Guide)
- [HowTo](https://schema.org/HowTo)
- [HowToDirection](https://schema.org/HowToDirection)
- [HowToSection](https://schema.org/HowToSection)
- [HowToStep](https://schema.org/HowToStep)
- [HowToTip](https://schema.org/HowToTip)
- [HyperToc](https://schema.org/HyperToc)
- [HyperTocEntry](https://schema.org/HyperTocEntry)
- [LearningResource](https://schema.org/LearningResource)
- [Legislation](https://schema.org/Legislation)
- [Manuscript](https://schema.org/Manuscript)
- [Map](https://schema.org/Map)
- [MathSolver](https://schema.org/MathSolver)
- [MediaObject](https://schema.org/MediaObject)
- [MediaReviewItem](https://schema.org/MediaReviewItem)
- [Menu](https://schema.org/Menu)
- [MenuSection](https://schema.org/MenuSection)
- [Message](https://schema.org/Message)
- [Movie](https://schema.org/Movie)
- [MusicComposition](https://schema.org/MusicComposition)
- [MusicPlaylist](https://schema.org/MusicPlaylist)
- [MusicRecording](https://schema.org/MusicRecording)
- [Painting](https://schema.org/Painting)
- [Photograph](https://schema.org/Photograph)
- [Play](https://schema.org/Play)
- [Poster](https://schema.org/Poster)
- [PublicationIssue](https://schema.org/PublicationIssue)
- [PublicationVolume](https://schema.org/PublicationVolume)
- [Quotation](https://schema.org/Quotation)
- [Review](https://schema.org/Review)
- [Sculpture](https://schema.org/Sculpture)
- [SheetMusic](https://schema.org/SheetMusic)
- [ShortStory](https://schema.org/ShortStory)
- [SoftwareApplication](https://schema.org/SoftwareApplication)
- [SoftwareSourceCode](https://schema.org/SoftwareSourceCode)
- [SpecialAnnouncement](https://schema.org/SpecialAnnouncement)
- [Statement](https://schema.org/Statement)
- [TVSeason](https://schema.org/TVSeason)
- [TVSeries](https://schema.org/TVSeries)
- [Thesis](https://schema.org/Thesis)
- [VisualArtwork](https://schema.org/VisualArtwork)
- [WebContent](https://schema.org/WebContent)
- [WebPage](https://schema.org/WebPage)
- [WebPageElement](https://schema.org/WebPageElement)
- [WebSite](https://schema.org/WebSite)

**Comment by joewhitsitt** [comment]
@mjoneill thanks for responding. I had a feeling I was going to open something up big with my question so I edited my comment to try to limit it a bit by specifying Article. That didn't work. ;)

>How can I best see what elements are available to us in the module to use?

We have the schema article functionality turned on (nested within the Article's SEO Settings). I'd start there if you want to look around. There are many open-ended types to extend it.

I guess the point I want to make is whether or not this should be fielded or just captured as part of the body along with adding to schema.

**Comment by mjoneill** [comment]
> We have the schema article functionality turned on (nested within the Article's SEO Settings). I'd start there if you want to look around. There are many open-ended types to extend it.

Yep, I've had a look at that. Should I assume that what I can see there as an editor(webmaster) is the full extent of Schema available to us for the Article type?

**Comment by joewhitsitt** [comment]
Ah. Yes, the schema.org: article dropdown has been restricted. Missing for webmasters is:

- aggregateRating
- dateModified
- datePublished
- hasPart
- mainEntityOfPage
- publisher
- review
- speakable

Some of these have the types that you have mentioned to make it more specific.

https://git.drupalcode.org/project/schema_metatag/-/blob/8.x-2.x/schema_article/config/schema/schema_article.metatag_tag.schema.yml

**Comment by pyrello** [comment]
I'm going to handle this via a site split for 91st Meridian. Hopefully we can get this considered as a platform feature down the road.
